presto-main/src/main/java/com/facebook/presto/FullConnectorSession.java

```diff
@@ -74,6 +78,27 @@ public FullConnectorSession(
         this.sessionPropertyManager = requireNonNull(sessionPropertyManager, "sessionPropertyManager is null");
         this.sqlFunctionProperties = session.getSqlFunctionProperties();
         this.sessionFunctions = ImmutableMap.copyOf(session.getSessionFunctions());
+        this.warningCollector = Optional.empty();
```

Comment:
We probably don't want to put warningCollector half way through the session. The session should be built with warningCollector from step 0. The place to put warningCollector into session is in `DispatchManager` where session was first created:
```java
session = sessionSupplier.createSession(queryId, sessionContext);
```

Note that `warningCollector` was created right after it because it depends on session. So in this case, you might need to change `createSession(queryId, sessionContext)` to be `createSession(queryId, sessionContext, warningCollectorFactory)` so that we create both session and warningCollector in one place.

Also, the next line `preparedQuery = queryPreparer.prepareQuery(session, query, warningCollector);` can also be simplified with `preparedQuery = queryPreparer.prepareQuery(session, query);` because warningCollector is now available in session already.

Reply:
> The session should be built with warningCollector from step 0.

@highker I checked the code and not sure how to deal with the warningCollector when converting Session to SessionRepresentation.
I see that SessionRepresentation is a thrift struct, is it possible to put the warningCollector to the SessionRepresentation too, or it doesn't matter and we do not need to pass warningCollector into SessionRepresentation?

Reply:
You can just use NOOP for session representation. Session representation is used for communication. We don't need warning collectors on the worker side. Make sure you have a comment on "we use NOOP to create a session from the representation as worker does not require warning collectors".

Reply:
> you might need to change createSession(queryId, sessionContext) to be createSession(queryId, sessionContext, warningCollectorFactory) so that we create both session and warningCollector in one place.

Done

> Also, the next line preparedQuery = queryPreparer.prepareQuery(session, query, warningCollector); can also be simplified with preparedQuery = queryPreparer.prepareQuery(session, query); because warningCollector is now available in session already.

I leave it as it is for now. As changing this function will lead to changes in  a bunch of callsites of this function. Since we are to add a separate PR to clean up the code with respect to warningCollector, I think it maybe better to leave it to the following PR.